src/pages-and-resources/discussions/app-config-form/apps/shared/DivisionByGroupFields.jsx

```diff
@@ -1,61 +1,142 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Form, TransitionReplace } from '@edx/paragon';
+import { FieldArray, useFormikContext } from 'formik';
 import FormSwitchGroup from '../../../../../generic/FormSwitchGroup';
 import messages from './messages';
 import AppConfigFormDivider from './AppConfigFormDivider';
+import { updateDividedDiscussionsIds } from '../../../data/slice';
+import { updateModel } from '../../../../../generic/model-store';
 
 function DivisionByGroupFields({
-  onBlur,
-  onChange,
-  intl,
-  values,
+  onBlur, onChange, intl, appConfig,
 }) {
+  const dispatch = useDispatch();
+  const { setFieldValue } = useFormikContext();
+  const {
+    dividedDiscussionsIds,
+    discussionTopics,
+    divideByCohorts,
+    divideCourseTopics,
+  } = appConfig;
+
+  useEffect(() => {
+    dispatch(updateDividedDiscussionsIds({ dividedDiscussionsIds }));
+  }, [dividedDiscussionsIds]);
+
+  useEffect(() => {
+    const discussionTopicIds = discussionTopics.map(
+      (topic) => topic.id,
+    );
+    const divideCourseTopicsSwitchOff = (
+      discussionTopicIds.length === dividedDiscussionsIds.length
+      && discussionTopicIds.every((topicId) => dividedDiscussionsIds.includes(topicId))
+    ) || !dividedDiscussionsIds.length;
```

Comment:
What does this check?

Reply:
It checks 2 cases. 
1- if dividedDiscussionsIds  is empty its means user just enable divide by cohorts switch. So divide Course wide Topics switch should be OFF.
2- if all discussion topic are divided so we need check it actually dividedDiscussionsIds have all discussion topics or not. If not then switch should be ON.

Reply:
got it